### PR TITLE
Basic in-memory implementations

### LINKED ISSUE #230: Cronus Sample throws an exception
This SimpleSample project throws an exception ``'Unable to resolve service for type 'Elders.Cronus.TypeContainer`1[Elders.Cronus.EventStore.Index.IEventStoreIndex]' while attempting to activate 'Elders.Cronus.EventStoreIndicesStartup'.'`` on startup.

### PR DESCRIPTION
**Pull request by ethno2405**
# Title
In memory implementations

## Related Issue 
closes #230
#215 

### Description
Basic in-memory implementation

### CODE REVIEW

**Comment by ethno2405** [comment]
@mynkow Updating Cronus projections and persistence with this changes might require adjusting their discoveries

File: src/Elders.Cronus/Discoveries/EventStoreDiscoveryBase.cs [hunk 1]
```diff
@@ -14,7 +17,26 @@ protected override DiscoveryResult<IEventStore> DiscoverFromAssemblies(Discovery
         IEnumerable<DiscoveredModel> GetModels(DiscoveryContext context)
         {
             var loadedTypes = context.Assemblies.Find<IEventStoreIndex>();
+            yield return new DiscoveredModel(typeof(EventToAggregateRootId), typeof(EventToAggregateRootId), ServiceLifetime.Transient);
```
Comment (mynkow) [inline src/Elders.Cronus/Discoveries/EventStoreDiscoveryBase.cs:20]: I would suggest to register all indices here. You could use something like:
```
foreach (var indexType in loadedTypes)
            {
                yield return new DiscoveredModel(indexType, indexType, ServiceLifetime.Singleton);
            }
```

I am 50/50 for the lifestyle singleton/transient. It is safer to go with Transient but leave a comment that we need to revisit it when we add more indices.

File: src/Elders.Cronus/Discoveries/EventStoreDiscoveryBase.cs [hunk 2]
```diff
@@ -14,7 +17,26 @@ protected override DiscoveryResult<IEventStore> DiscoverFromAssemblies(Discovery
...
+            var inMemoryModels = GetModels(context);
+            return new DiscoveryResult<IEventStore>(baseResult.Models.Concat(inMemoryModels));
+        }
+
+        IEnumerable<DiscoveredModel> GetModels(DiscoveryContext context)
+        {
+            yield return new DiscoveredModel(typeof(IEventStore), typeof(InMemoryEventStore), ServiceLifetime.Transient);
+            yield return new DiscoveredModel(typeof(InMemoryEventStoreStorage), typeof(InMemoryEventStoreStorage), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(IIndexStore), typeof(InMemoryIndexStore), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(IEventStorePlayer), typeof(InMemoryEventStorePlayer), ServiceLifetime.Singleton);
```
Comment (mynkow) [inline src/Elders.Cronus/Discoveries/EventStoreDiscoveryBase.cs:39]: We need to verify if all these interfaces are overridden by the persistence and projections packages. There is a boolean flag .CanOverride which has to be set to true

File: src/Elders.Cronus/Discoveries/ProjectionsDiscovery.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            yield return new DiscoveredModel(typeof(ProjectionRepository), typeof(ProjectionRepository), ServiceLifetime.Transient);
+            yield return new DiscoveredModel(typeof(IProjectionStore), typeof(InMemoryProjectionStore), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(IInitializableProjectionStore), typeof(NotInitializableProjectionStore), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(ISnapshotStore), typeof(InMemorySnapshotStore), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(ISnapshotStrategy), typeof(NoSnapshotStrategy), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(ProjectionRepositoryWithFallback<>), typeof(ProjectionRepositoryWithFallback<>), ServiceLifetime.Transient);
+            yield return new DiscoveredModel(typeof(ProjectionRepositoryWithFallback<,>), typeof(ProjectionRepositoryWithFallback<,>), ServiceLifetime.Transient);
+            yield return new DiscoveredModel(typeof(InMemoryProjectionVersionStore), typeof(InMemoryProjectionVersionStore), ServiceLifetime.Singleton);
```
Comment (mynkow) [inline src/Elders.Cronus/Discoveries/ProjectionsDiscovery.cs:32]: We need to verify if all these interfaces are overridden by the persistence and projections packages. There is a boolean flag .CanOverride which has to be set to true

**Pull request closed without merging**